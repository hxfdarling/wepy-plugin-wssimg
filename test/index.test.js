import { rollup } from 'rollup';

import url from '../';

process.chdir(__dirname);

describe('rollup-plugin-image', () => {
  it('use base64', () =>
    run('./fixtures/png.js', {}).then((bundle) => {
      expect(bundle.modules[1].code).toMatchSnapshot();
    }));
});

const DEFAULT_OPTIONS = {
  publicPath: '',
  emitFiles: true,
};

function run(input, options, experimentalCodeSplitting = false) {
  return rollup({
    input,
    plugins: [url(Object.assign({}, DEFAULT_OPTIONS, options))],
    experimentalCodeSplitting,
  });
}
