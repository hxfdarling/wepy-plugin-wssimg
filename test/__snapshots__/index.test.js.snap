// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-image use base64 1`] = `"var img = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGP6DwABBQECz6AuzQAAAABJRU5ErkJggg=='; export default img;"`;
